test/helper.js

```diff
@@ -3,58 +3,42 @@ var data        = require('../hooks/data');
 var taskcluster = require('taskcluster-client');
 var taskcreator = require('../hooks/taskcreator');
 var v1          = require('../routes/v1');
-var bin = {
-  server:             require('../bin/server')
-};
+var load        = require('../bin/main');
+var config      = require('typed-env-config');
+var _           = require('lodash');
 
-var testProfile = 'test';
+var loadOptions = {profile: 'test', process: 'test-helper'};
+var cfg = config({profile: 'test'});
```

Comment:
If I'm reading your middle comment correctly, I'm hearing you say that the elegant DI mock, which works perfectly well and isn't dependent on external services, should be replaced with something that is "ugly" and has all the downsides of integration tests?  That doesn't make a lot of sense.  In this case, yes, writing unit tests was *considerably* easier than setting up the "ugly" mechanics you suggest, and depending on pulse (which is known to be flaky) for my tests.

You'd make a much stronger case if all of the taskcluster components had test that passed in the github-linked CI and for which contributors could easily run the tests.  Until that point, I'm going to remain firmly unconvinced.